### DISCUSSION TITLE
qt client: the channel tree events are still prosece even when you are in another place, it wouldn't be a problem, but it causes an annoying problem.

### DISCUSSION THREAD

**Discussion by amirmahdifard** [BearWare/TeamTalk5#2008] [Ideas]
### Description

the annoying problem for this is when you are focused on something in the channel tree, and then you go to another place for example, in the files tree, or message input, or anyware else than channel tree, whenever the item in the channel tree gets updated, your focus will be taken off from the place that you were in, and will take you to the channel tree again: you will understand this more in the steps to reproduce section: vertion: 5.14

### Application

- [X] qtTeamTalk
- [ ] TeamTalkAndroid
- [ ] iTeamTalk
- [ ] TeamTalkClassic
- [ ] TeamTalkServer

### Platform

- [X] Windows
- [ ] macOS
- [ ] Android
- [ ] iOS
- [ ] Linux

### Expected behavior

when you are focused on another place, and the items in channel tree gets updated, it shouldn't interupt your focus, and it shouldn't focus you in channel tree.

### Actual behavior

when you are focused on another place and the items in channel tree gets updated, it takes your focus to the channel tree, and interupts your work

### Steps to reproduce problem

note: to be able to reproduce this problem properly, you need 2 users to do that, it's not properly reproduceable with only you alone, you will know why now.
1: join a channel, and tell another person to join to you.
2: it's not sometimes reproduceable with muting and unmuting microphone because i don't know why muting and unmuting microphone is not having any state change for the screen reader, but it will defenetly work with modifying status message: so, tell that person to put a status message for them selves.
3: in the channels tree, you focus your screen reader to that person's name.
4: tab or shift tab, and go to the message input, or the files list as an example places.
5: tell that person to modify or remove their status message.
you will see that your focus from files list or message input or anyware will be interupted and you will be put on channels tree again.
i hope you understood this issue, and i don't know why muting and unmuting microphone is not having any state change, but adding or removing status message does: the muting and unmuting microphone should also have state change like as status message does, but as mentioned abuv, this issue should be fixed and channel tree item updates shouldn't interupt your focus if you are anyware else than channel tree.

**Comment by amirmahdifard** [comment]
i should also note that this issue is not only started in 5.14, in older vertions too whenever an item in channel tree has been updated would interupt your focus if you were focusing anyware else than channels tree in the teamtalk window: it would be nice if this bug gets fixed.

**Comment by amirmahdifard** [comment]
@bear101 i think i explaned this bug clear enough, please explane me what's the problem? thanks.

**Comment by muamel1** [comment]
I confirm this and it is really annoying

**Comment by muamel1** [comment]
and also this #1858

**Comment by amirmahdifard** [comment]
thanks for your confurmation, may be if my issues get agreed with more people than only me alone, @bear101 will implement them: if you guys go look at list of my closed issues, you will clearly see how many good issues i opened that if all of them got implemented, teamtalk would be 1000 times bettar than all other apps: i, alone, found and reported all, and all of qt client's accessibility problems, and i also mentioned that if you want to remove the classic client, you should fix these at the very least: and when the classic client has been removed, you all saw that people are all complaning about the problems i reported befor: and i gave a lot of cool suggestions, and still when i look at them, i get sad because they were very cool and never got implemented.
